nan.h

```diff
@@ -801,44 +801,83 @@ class TryCatch {
 #endif
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::Function> func
     , int argc
     , v8::Local<v8::Value>* argv) {
+    v8::Isolate *isolate = v8::Isolate::GetCurrent();
+    v8::EscapableHandleScope scope(isolate);
+    v8::Local<v8::Object> target_obj;
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
-    return New(node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv));
+    if (target->IsUndefined() || target->IsNull()) {
+      target_obj = isolate->GetCurrentContext()->Global();
```

Comment:
Here's the 4.6 implementation of `Function::Call`, it seems it explicitly converts to an (internal) `Object` using `Utils::OpenHandle`.
https://github.com/nodejs/node/blob/master/deps/v8/src/api.cc#L4387

`Utils::OpenHandle` is here
https://github.com/nodejs/node/blob/master/deps/v8/src/api.h#L390-L398

But I don't really get how that `->IsX` check would pass. Are `v8::Undefined` and `v8::Null` both represented as a `nullptr` inside a `Local<>`?

Would it be possible to just hack this by doing a `reinterpret_cast<v8::Local<v8::Object>*>(&target)` and get V8 to do the right thing?